### PR TITLE
Clean up small release guide nits and typos

### PR DESCRIPTION
**Pull request by jrmccluskey**
Adds an explicit callout to not merge the website update PR and cleans up the example docker pulls to match how we actually format tags (no underscore before the `rcX` suffix.)

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by jrmccluskey** [comment]
Run Website_Stage_GCS PreCommit

**Comment by jrmccluskey** [comment]
Run Website PreCommit

**Comment by jrmccluskey** [comment]
R: @damccorm

File: website/www/site/content/en/contribute/release-guide.md [hunk 1]
```diff
@@ -532,6 +530,7 @@ help with this step. Please email `dev@` and ask a member of the `beammaintainer
   1. Create a PR to update beam-site, changes includes:
      * Copy python doc into beam-site
      * Copy java doc into beam-site
+     * **NOTE**: Do not merge this PR until after an RC has been approved.
```
Comment (damccorm) [inline website/www/site/content/en/contribute/release-guide.md:533]:
```suggestion
     * **NOTE**: Do not merge this PR until after an RC has been approved (see "Finalize the Release").
```

Optional addition to make it clear that people don't need to keep track of this themselves

**Review by damccorm — APPROVED** [review]
LGTM, thanks!

**Pull request merged**